resize.js

```diff
@@ -0,0 +1,76 @@
+var preserve_ratio = "%(preserve_ratio)s"; // string
+var min_height = %(min_height)d;
+var min_width = %(min_width)d;
+var resizable_style = "%(resizable_style)s";
+
+function resizeImage(idx, img){
+    $resizeImage($(img));
+}
+
+async function $resizeImage($img){
+    var img = $img[0];
+    while (img.naturalWidth == 0 || img.naturalHeight == 0) {
+        await new Promise(r => setTimeout(r, 1000));
+    }
+    var preserve_ratio_in_resizable = false;
+    if (preserve_ratio == "current" || preserve_ratio == "original"){
+        preserve_ratio_in_resizable = true;
+    }
+    if ($img.resizable("instance") == undefined ) {
+        $maybe_remove_a_dimension($img);
+        minHeight = ((min_height == null) ? 0: min_height)
+        minWidth = ((min_width == null) ? 0: min_width)
+        $img.resizable({
+            aspectRatio: preserve_ratio_in_resizable,
+            minHeight: minHeight,
+            minWidth: minWidth
+        });
+        $img.css("max-width", "100%%"); //%% because a single percent would make a python error during formatting of this file.
+        $img.dblclick(onDblClick);
+        $divUi = $img.parents("div[class^=ui-]");
+        $divUi.attr("contentEditable", "false");
+        $divUi.css("display", "inline-block");
+    } else {
+        console.log("Trying to apply resizable to image already resizable.");
+    }
+}
+
+function onDblClick(){
+    img = this;
```

Comment:
Declaring variables (especially with a common name) in the global namespace might lead to conflict with other poorly written js code using the same variable.

Reply:
oops, I forgot the difference between python and js. I had a var immediately